src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  get isAlive() {
+    return this.health > 0;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health <= 0) {
+      Animal.cleanupDeadAnimals();
+    }
+  }
 }
 
+Animal.alive = [];
+Animal.cleanupDeadAnimals = () => {
+  Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+};
```

Comment:
[CODE PERFORMANCE] - The cleanupDeadAnimals function uses the filter method, which is good as it avoids nesting loops. However, ensure that the cleanupDeadAnimals function is not called excessively, as it could lead to performance issues if the Animal.alive array becomes large. Consider calling it only when necessary, for example, after an animal's health is updated and the animal dies.